### PR TITLE
[Deployment] Comprehensive Update for K8s Yaml and Scripts

### PR DESCRIPTION
**Pull request by kevin-zhonghao**
This PR updates the K8s deployment yaml and scripts and fixes miscellaneous issues in API GW/DPM/RM/SM.

### CODE REVIEW

File: kubernetes/services/api-gateway.yaml [hunk 1]
```diff
@@ -43,6 +38,18 @@ data:
...
+    logging.level.root=TRACE
+    
+    Spring.mvc.async.request-timeout=30000
+    server.max-http-header-size=65536
+    server.max-initial-line-length=65536
+    
+    spring.sleuth.sampler.probability=1.0
+    spring.application.name=api-gateway
+    # the zipkin host should be YOURS
+    spring.zipkin.base-url=http://10.213.43.187:9411
```
Comment (xieus) [inline kubernetes/services/api-gateway.yaml:51]: @kevin-zhonghao @cj-chung Do we have zipkin conf in the application.properties files? We may need to remove them there and in the yaml files as well.

File: kubernetes/services/api-gateway.yaml [hunk 2]
```diff
@@ -4,23 +4,19 @@ metadata:
   name: api-configmap
 data:
   application.properties: |
+    ignite.thin.client.enable=true
```
Comment (xieus) [inline kubernetes/services/api-gateway.yaml:7]: Let us move the config back to where it was, along with other Ignite configs.

File: kubernetes/services/api-gateway.yaml [hunk 3]
```diff
@@ -54,7 +61,7 @@ spec:
   selector:
     matchLabels:
       app: apimanager
-  replicas: 5
+  replicas: 10
```
Comment (xieus) [inline kubernetes/services/api-gateway.yaml:64]: This is pretty aggressive :-) 5 is good enough.

File: kubernetes/services/port_manager.yaml [hunk 4]
```diff
@@ -7,23 +7,26 @@ data:
...
     microservices.vpc.service.url=http://vpcmanager-service.default.svc.cluster.local:9001
     microservices.subnet.service.url=http://subnetmanager-service.default.svc.cluster.local:9002
     microservices.ip.service.url=http://ipmanager-service.default.svc.cluster.local:9004/ips
     microservices.mac.service.url=http://macmanager-service.default.svc.cluster.local:9005/macs
     microservices.sg.service.url=http://sgmanager-service.default.svc.cluster.local:9008
     microservices.route.service.url=http://routemanager-service.default.svc.cluster.local:9003/routes
+    microservices.router.service.url=http://routemanager-service.default.svc.cluster.local:9003
     microservices.node.service.url=http://nodemanager-service.default.svc.cluster.local:9007/nodes
-    microservices.dataplane.service.url=http://dataplanemanager-service.default.svc.cluster.local:9010/network-configuration
+    microservices.dataplane.service.url=http://dataplanemanager-service.default.svc.cluster.local:9010/v4/port/
```
Comment (xieus) [inline kubernetes/services/port_manager.yaml:21]: @kevin-zhonghao @cj-chung We may still use V1 APIs in the Medina test cluster. Check this out.

**Review by xieus — CHANGES_REQUESTED** [review]
@kevin-zhonghao Like the title. It is very comprehensive 👍

**Review by cj-chung — CHANGES_REQUESTED** [review]

File: kubernetes/services/api-gateway.yaml (hunk 1 above)
Comment (kevin-zhonghao) [reply kubernetes/services/api-gateway.yaml:51]: We dont have zipkin conf in the application.properties files. It maybe added by Xiaodong

File: kubernetes/services/api-gateway.yaml (hunk 3 above)
Comment (kevin-zhonghao) [reply kubernetes/services/api-gateway.yaml:64]: Sure

File: kubernetes/services/api-gateway.yaml (hunk 1 above)
Comment (kevin-zhonghao) [reply kubernetes/services/api-gateway.yaml:51]: I have removed them

File: kubernetes/services/port_manager.yaml (hunk 4 above)
Comment (kevin-zhonghao) [reply kubernetes/services/port_manager.yaml]: Sure

File: kubernetes/services/api-gateway.yaml (hunk 2 above)
Comment (kevin-zhonghao) [reply kubernetes/services/api-gateway.yaml]: Done

File: kubernetes/services/api-gateway.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (kevin-zhonghao) [reply kubernetes/services/api-gateway.yaml]: have done

**Review by xieus — APPROVED** [review]
LGTM.

File: kubernetes/services/api-gateway.yaml [hunk 5]
```diff
@@ -5,22 +5,18 @@ metadata:
...
-    management.endpoints.web.exposure.include=health,prometheus
     logging.level.org.springframework.web=DEBUG
     logging.level.org.springframework.cloud=DEBUG
     logging.level.com.github.tomakehurst.wiremock=TRACE
-    logging.level.root=DEBUG
-
-    ignite.thin.client.enable=true
+    management.endpoints.web.exposure.include=health,prometheus
+    spring.sleuth.sampler.probability=1.0
+    spring.sleuth.web.skipPattern=(^health.*)
```
Comment (cj-chung) [inline kubernetes/services/api-gateway.yaml:13]: @kevin-zhonghao  Please remove them.

File: kubernetes/services/api-gateway.yaml [hunk 6]
```diff
@@ -43,6 +38,16 @@ data:
...
     keystone.auth_url=http://192.168.10.10:5000/v3
+    neutron.url_prefix=/v2.0
+    logging.level.root=TRACE
+    
+    Spring.mvc.async.request-timeout=30000
+    server.max-http-header-size=65536
+    server.max-initial-line-length=65536
+    
+    spring.sleuth.sampler.probability=1.0
+    spring.application.name=api-gateway
```
Comment (cj-chung) [inline kubernetes/services/api-gateway.yaml:49]: @kevin-zhonghao Please remove `spring.sleuth.sampler` and `spring.application.name`.

File: kubernetes/services/dpm_manager.yaml [hunk 7]
```diff
@@ -6,6 +6,36 @@ data:
   application.properties: |
     dataplane.grpc.port = 50001
     dataplane.isovs=true
+    spring.sleuth.sampler.probability=1.0
+    spring.application.name=dataplane-service
+    # the zipkin host should be YOURS
+    spring.zipkin.base-url=http://10.213.43.187:9411
```
Comment (cj-chung) [inline kubernetes/services/dpm_manager.yaml:12]: @kevin-zhonghao please remove them.

File: kubernetes/services/elastic_ip_manager.yaml [hunk 8]
```diff
@@ -9,6 +9,11 @@ data:
     ignite.port=10800
     ignite.thin.client.enable=true
     management.health.redis.enabled=false
+    
+    spring.sleuth.sampler.probability=1.0
+    spring.application.name=eip-manager
+    # the zipkin host should be YOURS
+    spring.zipkin.base-url=http://10.213.43.187:9411
```
Comment (cj-chung) [inline kubernetes/services/elastic_ip_manager.yaml:16]: @kevin-zhonghao Please remove them.

File: kubernetes/services/mac_manager.yaml [hunk 9]
```diff
@@ -4,19 +4,22 @@ metadata:
...
     #ignite.trust-store-path=truststore.jks
     #ignite.trust-store-password=123456
     logging.level.root=DEBUG
-    macmanager.oui = AA-BB-CC
+    macmanager.oui = aa:bb:cc
     macmanager.pool.size = 100
     macmanager.retrylimit = 10
+    
+    spring.sleuth.sampler.probability=1.0
+    spring.application.name=mac-manager
```
Comment (cj-chung) [inline kubernetes/services/mac_manager.yaml:22]: @kevin-zhonghao Please remove them.

File: kubernetes/services/node_manager.yaml [hunk 10]
```diff
@@ -18,6 +18,9 @@ data:
     spring.servlet.multipart.file-size-threshold=2KB
     spring.servlet.multipart.max-file-size=200MB
     spring.servlet.multipart.max-request-size=215MB
+    
+    spring.sleuth.sampler.probability=1.0
+    spring.application.name=node-manager
```
Comment (cj-chung) [inline kubernetes/services/node_manager.yaml:23]: @kevin-zhonghao Please remove them

File: kubernetes/services/port_manager.yaml [hunk 11]
```diff
@@ -7,23 +7,24 @@ data:
...
+    microservices.router.service.url=http://routemanager-service.default.svc.cluster.local:9003
     microservices.node.service.url=http://nodemanager-service.default.svc.cluster.local:9007/nodes
     microservices.dataplane.service.url=http://dataplanemanager-service.default.svc.cluster.local:9010/network-configuration
     microservices.elasticip.service.url=http://eipmanager-service.default.svc.cluster.local:9011
-
     alcor.vif_type=ovs
+    Spring.mvc.async.request-timeout=30000
+    
+    spring.sleuth.sampler.probability=1.0
+    spring.application.name=port-manager
```
Comment (cj-chung) [inline kubernetes/services/port_manager.yaml:27]: @kevin-zhonghao please remove them.

File: kubernetes/services/private_ip_manager.yaml [hunk 12]
```diff
@@ -8,6 +8,9 @@ data:
     ignite.host=ignite-alcor-ip-service.ignite-alcor-ip.svc.cluster.local
     ignite.port=10801
     logging.level.root=DEBUG
+    
+    spring.sleuth.sampler.probability=1.0
+    spring.application.name=ip-manager
```
Comment (cj-chung) [inline kubernetes/services/private_ip_manager.yaml:13]: @kevin-zhonghao Please remove them.

File: kubernetes/services/quota_manager.yaml [hunk 13]
```diff
@@ -20,6 +20,9 @@ data:
     quota.defaults.security_group_rule=-1
     quota.defaults.subnet=-1
     quota.defaults.subnetpool=-1
+    
+    spring.sleuth.sampler.probability=1.0
+    spring.application.name=quota-manager
```
Comment (cj-chung) [inline kubernetes/services/quota_manager.yaml:25]: @kevin-zhonghao Please remove them.

File: kubernetes/services/route_manager.yaml [hunk 14]
```diff
@@ -4,13 +4,22 @@ metadata:
...
     ignite.port=10800
     logging.level.root=DEBUG
     logging.level.org.springframework.web=info
     logging.file.path=.
     logging.type=file
+    management.health.redis.enabled=false
+    
+    spring.sleuth.sampler.probability=1.0
+    spring.application.name=route-manager
+
```
Comment (cj-chung) [inline kubernetes/services/route_manager.yaml:22]: @kevin-zhonghao please remove them.

File: kubernetes/services/sg_manager.yaml [hunk 15]
```diff
@@ -8,6 +8,9 @@ data:
     ignite.host=ignite-alcor-service.ignite-alcor.svc.cluster.local
     ignite.port=10800
     logging.level.root=DEBUG
+    
+    spring.sleuth.sampler.probability=1.0
+    spring.application.name=sg-manager
```
Comment (cj-chung) [inline kubernetes/services/sg_manager.yaml:13]: @kevin-zhonghao please remove them.

File: kubernetes/services/subnet_manager.yaml [hunk 16]
```diff
@@ -16,6 +16,12 @@ data:
     microservices.mac.service.url=http://macmanager-service.default.svc.cluster.local:9005/macs
     microservices.route.service.url=http://routemanager-service.default.svc.cluster.local:9003/
     microservices.ip.service.url=http://ipmanager-service.default.svc.cluster.local:9004/ips/
+    microservices.port.service.url=http://portmanager-service.default.svc.cluster.local:9006/
+    
+    spring.sleuth.sampler.probability=1.0
+    spring.application.name=subnet-manager
```
Comment (cj-chung) [inline kubernetes/services/subnet_manager.yaml:22]: @kevin-zhonghao Please remove them

File: kubernetes/services/vpc_manager.yaml [hunk 17]
```diff
@@ -12,8 +12,12 @@ data:
     logging.file.path=.
     logging.type=file
     apache.kafka.address=172.17.0.1:9092
-    microservices.route.service.url=http://routemanager-service.default.svc.cluster.local:9003/vpcs/
+    microservices.route.service.url=http://routemanager-service.default.svc.cluster.local:9003/
     microservices.subnet.service.url=http://subnetmanager-service.default.svc.cluster.local:9002
+    
+    rbac.policy.type=Enforced
+    spring.sleuth.sampler.probability=1.0
+    spring.application.name=vpc-manager
```
Comment (cj-chung) [inline kubernetes/services/vpc_manager.yaml:20]: @kevin-zhonghao please remove them.

File: web/src/main/java/com/futurewei/alcor/web/entity/subnet/SubnetEntity.java [hunk 18]
```diff
@@ -63,8 +63,8 @@
     @JsonProperty("secondary_dns")
     private String secondaryDns;
 
-    @JsonProperty("routes")
-    private List<RouteEntity> routeEntities;
+//    @JsonProperty("routes")
+//    private List<RouteEntity> routeEntities;
```
Comment (cj-chung) [inline web/src/main/java/com/futurewei/alcor/web/entity/subnet/SubnetEntity.java:67]: @kevin-zhonghao Remove it if not using.

**Review by cj-chung — CHANGES_REQUESTED** [review]

File: kubernetes/services/api-gateway.yaml [hunk 19]
```diff
@@ -4,23 +4,19 @@ metadata:
...
-    management.endpoints.web.exposure.include=health,prometheus
     logging.level.org.springframework.web=DEBUG
     logging.level.org.springframework.cloud=DEBUG
     logging.level.com.github.tomakehurst.wiremock=TRACE
-    logging.level.root=DEBUG
-
-    ignite.thin.client.enable=true
+    management.endpoints.web.exposure.include=health,prometheus
+    spring.sleuth.sampler.probability=1.0
+    spring.sleuth.web.skipPattern=(^health.*)
```
[the remark this answers is not in the document]
Comment (kevin-zhonghao) [reply kubernetes/services/api-gateway.yaml:13]: Already removed

File: web/src/main/java/com/futurewei/alcor/web/entity/subnet/SubnetEntity.java (hunk 18 above)
Comment (kevin-zhonghao) [reply web/src/main/java/com/futurewei/alcor/web/entity/subnet/SubnetEntity.java:67]: Sure

**Comment by kevin-zhonghao** [comment]
All comments have been solved. Ready to review again

**Review by cj-chung — APPROVED** [review]
LGTM

**Pull request merged**